core/src/main/java/ai/timefold/solver/core/api/score/buildin/hardmediumsoft/HardMediumSoftScore.java

```diff
@@ -24,29 +24,30 @@
  */
 public final class HardMediumSoftScore implements Score<HardMediumSoftScore> {
 
-    public static final @NonNull HardMediumSoftScore ZERO = new HardMediumSoftScore(0, 0, 0, 0);
-    public static final @NonNull HardMediumSoftScore ONE_HARD = new HardMediumSoftScore(0, 1, 0, 0);
-    private static final @NonNull HardMediumSoftScore MINUS_ONE_HARD = new HardMediumSoftScore(0, -1, 0, 0);
-    public static final @NonNull HardMediumSoftScore ONE_MEDIUM = new HardMediumSoftScore(0, 0, 1, 0);
-    private static final @NonNull HardMediumSoftScore MINUS_ONE_MEDIUM = new HardMediumSoftScore(0, 0, -1, 0);
-    public static final @NonNull HardMediumSoftScore ONE_SOFT = new HardMediumSoftScore(0, 0, 0, 1);
-    private static final @NonNull HardMediumSoftScore MINUS_ONE_SOFT = new HardMediumSoftScore(0, 0, 0, -1);
+    public static final @NonNull HardMediumSoftScore ZERO = new HardMediumSoftScore(0, 0, 0);
+    public static final @NonNull HardMediumSoftScore ONE_HARD = new HardMediumSoftScore(1, 0, 0);
+    private static final @NonNull HardMediumSoftScore MINUS_ONE_HARD = new HardMediumSoftScore(-1, 0, 0);
+    public static final @NonNull HardMediumSoftScore ONE_MEDIUM = new HardMediumSoftScore(0, 1, 0);
+    private static final @NonNull HardMediumSoftScore MINUS_ONE_MEDIUM = new HardMediumSoftScore(0, -1, 0);
+    public static final @NonNull HardMediumSoftScore ONE_SOFT = new HardMediumSoftScore(0, 0, 1);
+    private static final @NonNull HardMediumSoftScore MINUS_ONE_SOFT = new HardMediumSoftScore(0, 0, -1);
 
     public static @NonNull HardMediumSoftScore parseScore(@NonNull String scoreString) {
         String[] scoreTokens = ScoreUtil.parseScoreTokens(HardMediumSoftScore.class, scoreString,
                 HARD_LABEL, MEDIUM_LABEL, SOFT_LABEL);
-        int initScore = ScoreUtil.parseInitScore(HardMediumSoftScore.class, scoreString, scoreTokens[0]);
         int hardScore = ScoreUtil.parseLevelAsInt(HardMediumSoftScore.class, scoreString, scoreTokens[1]);
```

Comment:
Should we begin with index 0 instead? If so, I wonder why no test has failed.

Reply:
Good observation! Neither I nor the tests noticed this, because the code was actually written very resiliently. Even though the init score was no longer there, it still allocated the array as if it were - therefore the code didn't notice anything, just that the init level in the array was empty.

Anyway... It's fixed now and the code now only generates the arrays it actually needs. No init score.